packages/mui-base/src/useSelect/useSelect.ts

```diff
@@ -351,24 +370,42 @@ function useSelect<OptionValue, Multiple extends boolean = false>(
     externalProps: ExternalProps = {} as ExternalProps,
   ): UseSelectButtonSlotProps<ExternalProps> => {
     const externalEventHandlers = extractEventHandlers(externalProps);
-    const listboxAndButtonProps = combineHooksSlotProps(getButtonRootProps, getListboxRootProps);
-    const combinedProps = combineHooksSlotProps(listboxAndButtonProps, getSelectTriggerProps);
+    const propGetter =
+      focusManagement === 'activeDescendant'
+        ? combineHooksSlotProps(getButtonRootProps, getListboxRootProps)
+        : getButtonRootProps;
+
+    const combinedProps = combineHooksSlotProps(propGetter, getSelectTriggerProps);
+
     return {
       ...externalProps,
       ...combinedProps(externalEventHandlers),
     };
   };
 
+  const getActiveDescendantListboxProps = (otherHandlers: EventHandlers = {}) => {
+    return {
+      ref: handleListboxRef,
+      onMouseDown: preventDefault, // to prevent the button from losing focus when interacting with the listbox
+      ...otherHandlers,
+    };
+  };
+
   const getListboxProps = <ExternalProps extends Record<string, unknown>>(
     externalProps: ExternalProps = {} as ExternalProps,
   ): UseSelectListboxSlotProps<ExternalProps> => {
+    const externalEventHandlers = extractEventHandlers(externalProps);
+    const propGetter =
+      focusManagement === 'activeDescendant'
+        ? getActiveDescendantListboxProps
+        : getListboxRootProps;
+
     return {
```

Comment:
Here, I have the same issue as in [my other comment](https://github.com/mui/material-ui/pull/39675/files#r1376475392).